src/stats/stats_publisher.cpp

```diff
@@ -624,8 +624,18 @@ void vma_stats_instance_create_epoll_block(int fd, iomux_func_stats_t* local_sta
 void vma_stats_instance_remove_epoll_block(iomux_func_stats_t* local_stats_addr)
 {
 	g_lock_ep_stats.lock();
-        epoll_stats_t* ep_stats = (epoll_stats_t*)g_p_stats_data_reader->pop_p_skt_stats(local_stats_addr);
-        if (ep_stats)
-                ep_stats->enabled = false;
+	iomux_func_stats_t* ep_func_stats = (iomux_func_stats_t*)g_p_stats_data_reader->pop_p_skt_stats(local_stats_addr);
+	if (ep_func_stats) {
+		// Search ep_mem block to release
+		for (int i=0; i<NUM_OF_SUPPORTED_EPFDS; i++) {
+			if (&g_sh_mem->iomux.epoll[i].stats == ep_func_stats) {
+				g_sh_mem->iomux.epoll[i].enabled = false;
+				g_lock_ep_stats.unlock();
+				return;
+			}
+		}
+	}
+
+	vlog_printf(VLOG_WARNING, "Can't remove epoll stat memory block\n");
```

Comment:
Need to follow the same pattern for removal of other components in the file. For example see 
void vma_stats_instance_remove_cq_block(cq_stats_t* local_stats_addr)